plugins/modules/digital_ocean_domain.py

```diff
@@ -162,6 +161,19 @@ def core(module):
             if 'message' in domain:
                 module.fail_json(changed=False, msg=domain['message'])
             else:
+                # We're at the mercy of a backend process which we have no visibility into:
+                # https://developers.digitalocean.com/documentation/v2/#create-a-new-domain
+                #
+                # In particular: "Keep in mind that, upon creation, the zone_file field will
+                # have a value of null until a zone file is generated and propagated through
+                # an automatic process on the DigitalOcean servers."
+                #
+                # Arguably, it's nice to see the records versus null, so, we'll just try a
+                # few times before giving up and returning null.
+                for i in range(5):
+                    if do_manager.domain_record()['domain']['zone_file'] != 'null':
```

Comment:
The API returns back a JSON `null`, which maps to `None` after the parsing is over.

```suggestion
                    if do_manager.domain_record()['domain']['zone_file'] is not None:
```